### PR TITLE
ENT-2310 | Adding a post_save receiver to listen to CourseEnrollments…

### PR DESCRIPTION
**Pull request by christopappas**
… table

~Before I merged, I definitely want to make sure I can verify that the async element of this will work on Production... will want to mess around a bit more with Devstack to that end.~

This works in devstack with `CELERY_ALWAYS_EAGER=True`, meaning that tasks run synchronously. Production and staging environment configs are different enough from devstack that I'm not 100% convinced testing asynchronous functionality means this code will or will not work in production, though I will still test setting up a broker and worker locally to prove the concept.

Barring discovering anything that drastically changes my approach, this should be ready for review!

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
- [ ] Version pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)

### CODE REVIEW

File: enterprise/tasks.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    Create enterprise enrollment for user if course_id part of catalog for the ENT customer.
+    """
+    enterprise_customer = enterprise_customer_user.enterprise_customer
+    if enterprise_customer.catalog_contains_course(course_id):
+        LOGGER.info((
+            "Creating EnterpriseCourseEnrollment for user %s "
+            "on course %s for enterprise_customer %s"
+        ), enterprise_customer_user.user_id, course_id, enterprise_customer)
+        EnterpriseCourseEnrollment.objects.create(
```
Comment (christopappas) [inline enterprise/tasks.py]: maybe use a `get_or_create` here just to be safe? (so we don't create duplicates)

Comment (christopappas) [reply enterprise/tasks.py]: clicking enroll in devstack did NOT create 2 enterprise course enrollments, so i do not think this is necessary

Comment (adamstankiewicz) [reply enterprise/tasks.py]: Hmm, good question. Even though in testing with devstack, it did not create 2 enterprise course enrollments... I feel that it might be good to be safe here just in case with `get_or_create`. If there are any edge cases where an enrollment might already exist in stage/production data, might be good to proactively handle that.

File: tests/test_enterprise/test_tasks.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    """
+    Tests tasks associated with Enterprise.
+    """
+    def setUp(self):
+        """
+        Setup for `TestEnterpriseTasks` test.
+        """
+        self.user = UserFactory(id=2, email='user@example.com')
+        self.enterprise_customer = EnterpriseCustomerFactory(
+            name='Team Titans',
```
Comment (adamstankiewicz) [inline tests/test_enterprise/test_tasks.py]: nice enterprise customer name ;)

**Comment by christopappas** [comment]
The task will not be kicked off if `created` is not True in the signal though. So it's not just a save that has to happen for this task to be run, but a create on the CourseEnrollment table

Would there ever be a case where there would be more than 1 course enrollment created that maps to a single enterpriseCourseEnrollment? If not, I think this would be fine, right?

**Comment by christopappas** [comment]
@brittneyexline 🔼

**Comment by brittneyexline** [comment]
i'm saying that if the EnterpriseCourseEnrollment record had already been created at the point that the task has been kicked off, something that is definitely possible for the current enterprise enrollment workflow (see https://github.com/edx/edx-enterprise/blob/master/enterprise/models.py#L760), then we could bypass doing the work of in the task in this case too.

**Comment by christopappas** [comment]
Ah, I see. Update coming

**Comment by christopappas** [comment]
@brittneyexline addressed your recommendation. updated to make sure we check if the EnterpriseCourseEnrollment exists before making the call to discovery (and added a test)
